### DISCUSSION TITLE
Multicam refresh fails

### DISCUSSION THREAD

**Discussion by vic4news** [gtxaspec/wz_mini_hacks#650] [Q&A] [answered]
@virmaior, I’ve started using the multicam web page on my iPad but the refresh fails and the cam boxes are outlined in red to indicate a failure. I’m using the current release of Wed Oct 25. 

Anyway to tell what is causing the refresh failure?

**Comment by virmaior** [comment]
I'll look into it!

I actually use a variation of the code rather than code itself on a raspberry pi.

Two immediate possibilities come to mind:
1. CORS 
2. Refresh rate is too high for the cameras to handle well. (on my local network it takes about 200ms / camera -- any faster and the cameras start crashing)

Editing the block of javascript in `index.html` to 
```
<script type="text/javascript">


window.onload = function()
{
  cam_tool.add_camera(1,'http://192.168.9.101/cgi-bin/jpeg.cgi?channel=0','v3');
  cam_tool.add_camera(2,'http://192.168.9.102/cgi-bin/jpeg.cgi?channel=0','v3');
  cam_tool.add_camera(3,'http://192.168.9.103/cgi-bin/jpeg.cgi?channel=0','v3');
  cam_tool.add_camera(4,'http://192.168.9.104/cgi-bin/jpeg.cgi?channel=0','v3');
  cam_tool.add_camera(5,'http://192.168.9.105/cgi-bin/jpeg.cgi?channel=0','v3');

  cam_tool.init(800);
};
</script>

```
(one of the improvements in this version is that the javascript to make it work is separted out as index.js whereas the javascript for settings is in index.html).
this works for displaying the five cameras and getting a few frames per second on my computer ...
marking which cameras are in night mode is getting blocked by CORS  (https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)...

can overcome CORS by allowing it (considered a bad security practice):
```
  echo "HTTP/1.1 200"
  echo -e "Access-Control-Allow-Origin: *"
  echo -e "Content-type: text/html\n\n"
  echo ""
```

(If you use a proxy, you don't get cors issues).

If you open up the javascript developer console and the network pane, are you seeing any notable errors?

**Reply by vic4news** [reply, marked as answer]
I downloaded web inspector for iPad and was able to see an error stating the url contained user credentials. I had tried setting WEB_SERVER_AUTH="false" earlier but still was using the credentials in the url. 
Once I removed the credentials from the URL refresh started working!

Thanks for your help
